engine/runtime/src/main/java/org/enso/interpreter/runtime/data/vector/VectorFromFunctionNode.java

```diff
@@ -0,0 +1,150 @@
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.profiles.BranchProfile;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.node.callable.dispatch.InvokeFunctionNode;
+import org.enso.interpreter.node.expression.builtin.error.AdditionalWarnings;
+import org.enso.interpreter.node.expression.builtin.error.NoWrap;
+import org.enso.interpreter.node.expression.builtin.error.ProblemBehavior;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.callable.function.Function;
+import org.enso.interpreter.runtime.data.atom.Atom;
+import org.enso.interpreter.runtime.error.DataflowError;
+import org.enso.interpreter.runtime.error.PanicException;
+import org.enso.interpreter.runtime.library.dispatch.TypesLibrary;
+import org.enso.interpreter.runtime.state.State;
+import org.enso.interpreter.runtime.warning.AppendWarningNode;
+import org.enso.interpreter.runtime.warning.Warning;
+import org.enso.interpreter.runtime.warning.WarningsLibrary;
+
+@BuiltinMethod(
+    type = "Array_Like_Helpers",
+    name = "vector_from_function",
+    description = "Creates a vector from a function.")
+public abstract class VectorFromFunctionNode extends Node {
+  public static VectorFromFunctionNode build() {
+    return VectorFromFunctionNodeGen.create();
+  }
+
+  private static final int MAX_MAP_WARNINGS = 10;
+
+  /**
+   * @param length Length of the vector to create.
+   * @param func Callback function called with index as argument.
+   * @param onProblems Can be either an atom of type {@code Problem_Behavior} or {@code No_Wrap}
+   *     type.
+   * @return Vector constructed from the given function.
+   */
+  abstract Object execute(
+      VirtualFrame frame, State state, long length, Function func, Object onProblems);
+
+  @Specialization
+  Object doIt(
+      VirtualFrame frame,
+      State state,
+      long length,
+      Function func,
+      Object onProblemsAtom,
+      @Cached("buildWithArity(1)") InvokeFunctionNode invokeFunctionNode,
+      @Cached("build()") AppendWarningNode appendWarningNode,
+      @CachedLibrary(limit = "3") WarningsLibrary warnsLib,
+      @CachedLibrary(limit = "3") TypesLibrary typesLib,
+      @Cached BranchProfile errorEncounteredProfile) {
+    var ctx = EnsoContext.get(this);
+    var onProblems = processOnProblemsArg(onProblemsAtom, typesLib);
+    var len = Math.toIntExact(length);
+    var nothing = ctx.getNothing();
+    var target = ArrayBuilder.newBuilder(len);
+    var errorsEncountered = 0;
+    for (int i = 0; i < len; i++) {
+      var value = invokeFunctionNode.execute(func, frame, state, new Long[] {(long) i});
+      Object valueToAdd = value;
+      if (value instanceof DataflowError err) {
+        errorEncounteredProfile.enter();
+        switch (onProblems) {
```

Comment:
It would help the size of IGV graph if `onProblems` was a constant when a single `DataflowError` appears.

Reply:
`onProblems` parameter is cached in 89b5e59f4d7fe38d063e4aeb598a138fc7aae350